Submodule OmniBLE contains modified content
diff --git a/OmniBLE/OmniBLE/PumpManagerUI/ViewControllers/DashUICoordinator.swift b/OmniBLE/OmniBLE/PumpManagerUI/ViewControllers/DashUICoordinator.swift
index 0fd0f68..5003df0 100644
--- a/OmniBLE/OmniBLE/PumpManagerUI/ViewControllers/DashUICoordinator.swift
+++ b/OmniBLE/OmniBLE/PumpManagerUI/ViewControllers/DashUICoordinator.swift
@@ -40,11 +40,11 @@ enum DashUIScreen {
         case .insulinTypeSelection:
             return .pairPod
         case .pairPod:
-            return .confirmAttachment
+            return .insertCannula
         case .confirmAttachment:
             return .insertCannula
         case .insertCannula:
-            return .checkInsertedCannula
+            return nil
         case .checkInsertedCannula:
             return .setupComplete
         case .setupComplete:
@@ -347,11 +347,7 @@ class DashUICoordinator: UINavigationController, PumpManagerOnboarding, Completi
         if pumpManager.state.podState?.needsCommsRecovery == true {
             return .pendingCommandRecovery
         } else if pumpManager.podCommState == .activating {
-            if pumpManager.podAttachmentConfirmed {
-                return .insertCannula
-            } else {
-                return .confirmAttachment
-            }
+            return .insertCannula
         } else if !pumpManager.isOnboarded {
             if !pumpManager.initialConfigurationCompleted {
                 return .firstRunScreen
