From 0d38c214bd06a5a850e0b651c2b94eed7a67c63d Mon Sep 17 00:00:00 2001
From: jonasbjo <jonas.bjorkert@tele2.com>
Date: Sun, 5 Feb 2023 15:21:51 +0100
Subject: [PATCH] proteinLine

---
 LoopFollow/Controllers/Graphs.swift | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/LoopFollow/Controllers/Graphs.swift b/LoopFollow/Controllers/Graphs.swift
index 9e29b90..89c725c 100644
--- a/LoopFollow/Controllers/Graphs.swift
+++ b/LoopFollow/Controllers/Graphs.swift
@@ -372,6 +372,13 @@ extension MainViewController {
                 BGChart.xAxis.addLimitLine(ul)
             }
         }
+
+        //Protein line
+        let ul3 = ChartLimitLine()
+        ul3.limit = Double(dateTimeUtils.getNowTimeIntervalUTC().advanced(by: -90 * 60))
+        ul3.lineColor = NSUIColor.systemOrange.withAlphaComponent(0.5)
+        ul3.lineWidth = 1
+        BGChart.xAxis.addLimitLine(ul3)
     }
     
     func createMidnightLines() {
-- 
2.38.1

